frontend/src/components/Trials/Trial.tsx

```diff
@@ -1,71 +1,125 @@
 import React, {useEffect, useState} from "react";
 import {useTrialAPI} from "../../apis";
 import Header from "../Header";
-import WrittenStrategy from "./WrittenStrategy";
-import ConsentForm from "./Intro/Consent";
-import Selection from "./SocialLearning/Selection";
-import IndividualTrial from "./IndividualTrial";
+import {Advisor, Solution, WrittenStrategy as WrittenStrategyApiTypes} from "../../apis/apiTypes";
+import WaitForNextTrialScreen from "./WaitForNextTrialScreen";
+import {renderTrial, setHeaderTitle} from "./RenderTrialComponent";
 
 interface TrialInterface {
     nextTrialHandler: () => null;
 }
 
 const Trial: React.FC<TrialInterface> = (props) => {
-    const {trialData, loading, error, axiosRequest} = useTrialAPI({method: 'GET'});
+    const {trial, loading, error, axiosGet, axiosPost} = useTrialAPI();
+
     const [trialType, setTrialType] = useState<string>('');
-    const [trialNumber, setTrialNumber] = useState<number>(0);
+    const [socialLearningType, setSocialLearningType] = useState<string>('');
+    const [waitingForTheNextTrial, setWaitingForTheNextTrial] = useState<boolean>(false);
+    const [teacherInx, setTeacherInx] = useState<number>();
+    const [learningExampleInx, setLearningExampleInx] = useState<number>(0);
+    // Total points of the player collected in the individual trials
+    const [totalPoints, setTotalPoints] = useState<number>(0);
 
     useEffect(() => {
-        setTrialType(trialData?.trial_type);
-        setTrialNumber(trialData?.trial_num_in_session);
-    }, [trialData]);
+        const t = JSON.parse(window.localStorage.getItem('teacherInx'));
+        if (t) setTeacherInx(t);
+        const e = JSON.parse(window.localStorage.getItem('learningExampleInx'));
+        if (e) setLearningExampleInx(e);
+        const p = JSON.parse(window.localStorage.getItem('totalPoints'));
+        if (p) setTotalPoints(p);
```

Comment:
In this case `teacherInx`, `learningExampleInx`, and `totalPoints` are only created and stored on the frontend side. You're right, I think it might be cleaner to keep them as part of the data model.